libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/PersistedStats.java

```diff
@@ -65,9 +75,15 @@ public synchronized void incrementSyncStats(long bytesSent, long reports, long c
                 observationsToday);
 
 
-        final long lastUploadMs = Long.parseLong(properties.getProperty(DataStorageContract.Stats.KEY_LAST_UPLOAD_TIME, "0"));
-        final int timeDiffSec = Long.valueOf((time - lastUploadMs) / 1000).intValue();
-        if (lastUploadMs > 0 && timeDiffSec > 0) {
+        Intent intent = new Intent();
```

Comment:
I think we should we check that newProps has at least some valid attributes before firing it off to the broadcast service. There's no sense in sending message that can't be parsed - and will cause all receivers to probably reset their write counters to 0.